editor/src/dev/load_project.clj

```diff
@@ -211,11 +211,13 @@
                 :update-successors
                 (it/update-successors transaction-context))
 
-              (it/trace-dependencies transaction-context)
-              (it/apply-tx-label transaction-context)
-              (it/finalize-update transaction-context))
+              (when transaction-context
+                (it/trace-dependencies transaction-context)
+                (it/apply-tx-label transaction-context)
+                (it/finalize-update transaction-context)))
 
-        _ (g/commit-tx-result! tx-result transact-opts)
+        _ (when tx-result
+            (g/commit-tx-result! tx-result transact-opts))
```

Comment:
These were wrapped in `when` to make stuff work OK when the `final-task` is defined to something earlier than the input they're expecting.